broadinstitute/single_cell_portal_core
# global gene search collapse genes to violin (SCP-2934, SCP-2933)

![image](https://user-images.githubusercontent.com/2800795/104235992-cde72c00-5423-11eb-8934-ff11b338c11f.png)

To test:
1. Populate synthetic study 'human_lymph_node' (and ensure `rails jobs:work` is running)
2. do global gene search for apoe and agpat2
3. Confirm dotplot appears, along with 'Collapse by genes' dropdown.
4. click label or (?) icon to confirm tooltip
5. select 'mean'
6. Observe the graph change to violin plot, with data reflecting cell averages.  Label under graph should now say 'mean expression of agpat2, apoe'
7. Do search for just apoe
8. confirm 'Collapse genes by' dropdown does not appear

We have an issue where the 'collapse_by' variable is called a bunch of different things across our codebase. 'consensus' 'collapse' 'row_centered' (the latter is from morpheus), etc...  When we fix the existing study explore collapse function, we should standardize as best we can.  (I'd lean towards 'collapse_by' because that's what's in the UI, but don't feel strongly)
